x-pack/plugins/apm/server/lib/services/get_services/get_services_items_stats.ts

```diff
@@ -0,0 +1,309 @@
+
+const getDeltaAsMinutes = (setup: ServicesItemsSetup) =>
+  (setup.end - setup.start) / 1000 / 60;
+
+interface AggregationParams {
+  setup: ServicesItemsSetup;
+  projection: ServicesItemsProjection;
+}
+
+export const getTransactionDurationAverages = async ({
+  setup,
+  projection,
+}: AggregationParams) => {
+  const { client, indices } = setup;
+
+  const response = await client.search(
+    mergeProjection(projection, {
+      size: 0,
+      index: indices['apm_oss.transactionIndices'],
+      body: {
+        query: {
+          bool: {
+            filter: projection.body.query.bool.filter.concat({
+              term: {
+                [PROCESSOR_EVENT]: ProcessorEvent.transaction,
+              },
+            }),
+          },
+        },
+        aggs: {
+          services: {
+            terms: {
+              ...projection.body.aggs.services.terms,
+              size: MAX_NUMBER_OF_SERVICES,
+            },
+            aggs: {
+              average: {
+                avg: {
+                  field: TRANSACTION_DURATION,
+                },
+              },
+            },
+          },
+        },
+      },
+    })
+  );
+
+  const { aggregations } = response;
+
+  if (!aggregations) {
+    return [];
+  }
+
+  return aggregations.services.buckets.map((bucket) => ({
+    serviceName: bucket.key as string,
+    avgResponseTime: bucket.average.value,
+  }));
+};
+
+export const getAgentNames = async ({
+  setup,
+  projection,
+}: AggregationParams) => {
+  const { client, indices } = setup;
+  const response = await client.search(
+    mergeProjection(projection, {
+      index: [
+        indices['apm_oss.metricsIndices'],
+        indices['apm_oss.errorIndices'],
+        indices['apm_oss.transactionIndices'],
+      ],
+      body: {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              ...projection.body.query.bool.filter,
+              {
+                terms: {
+                  [PROCESSOR_EVENT]: [
+                    ProcessorEvent.metric,
+                    ProcessorEvent.error,
+                    ProcessorEvent.transaction,
+                  ],
+                },
+              },
+            ],
+          },
+        },
+        aggs: {
+          services: {
+            terms: {
+              ...projection.body.aggs.services.terms,
+              size: MAX_NUMBER_OF_SERVICES,
+            },
+            aggs: {
+              agent_name: {
+                top_hits: {
+                  _source: [AGENT_NAME],
+                  size: 1,
+                },
+              },
+            },
+          },
+        },
+      },
+    })
+  );
+
+  const { aggregations } = response;
+
+  if (!aggregations) {
+    return [];
+  }
+
+  return aggregations.services.buckets.map((bucket) => ({
+    serviceName: bucket.key as string,
+    agentName: (bucket.agent_name.hits.hits[0]?._source as {
+      agent: {
+        name: string;
+      };
+    }).agent.name,
```

Comment:
Is this not the same?

```suggestion
    agentName: (bucket.agent_name.hits.hits[0]?._source as string
```